### PR TITLE
Add callback that freezes specified module

### PR DESCRIPTION
**Pull request by dxoigmn**
# What does this PR do?

Fixes #\<issue_number>

## Type of change

Please check all relevant options.

- [ ] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Testing

Please describe the tests that you ran to verify your changes. Consider listing any relevant details of your test configuration.

- [ ] Test A
- [ ] Test B

## Before submitting

- [ ] The title is **self-explanatory** and the description **concisely** explains the PR
- [ ] My **PR does only one thing**, instead of bundling different changes together
- [ ] I list all the **breaking changes** introduced by this pull request
- [ ] I have commented my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have run pre-commit hooks with `pre-commit run -a` command without errors

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: mart/callbacks/freeze.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                f"The LightningModule should have a nn.Module `{self.name}` attribute"
+            )
+
+        for name, param in module.named_parameters():
+            logger.debug(f"Disabling gradient for {name}")
+            param.requires_grad_(False)
+
+        for name, module in module.named_modules():
+            module_kind = module.__class__.__name__
+            if "BatchNorm" in module_kind:
```
Comment (dxoigmn) [inline mart/callbacks/freeze.py:42]: We should probably make this configurable. For example, YoloV4 uses DropBlock: https://github.com/AlexeyAB/Yet-Another-YOLOv4-Pytorch/blob/d80d6a20372598b6306b37218cb61533e8bd9592/model.py#L102.

File: mart/callbacks/freeze.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+class FreezeModule(Callback):
+    def __init__(
+        self,
+        module="backbone",
+    ):
+        self.name = module
+
+    def setup(self, trainer, pl_module, stage):
+        # FIXME: Use DotDict?
+        module = getattr(pl_module.model, self.name, None)
```
Comment (dxoigmn) [inline mart/callbacks/freeze.py:29]: Use attrgetter

**Review by mzweilin — COMMENTED** [review]
I think this is the right direction (not relying on .eval). I wonder if we can make this automated without specific configuration.

**Comment by dxoigmn** [comment]
I should note that I'm not sure this works in multi-gpu mode.

**Comment by dxoigmn** [comment]
> I should note that I'm not sure this works in multi-gpu mode.

This does work but one must beware that `BatchNorm` modules get turned into `SyncBatchNorm` when using DDP:
https://github.com/IntelLabs/MART/blob/ed89c722f8602885f738cd2765af3d3de97c10af/mart/configs/trainer/ddp.yaml#L8

File: mart/configs/callbacks/attack_in_eval_mode.yaml [hunk 3]
```diff
@@ -1,2 +1,11 @@
 attack_in_eval_mode:
   _target_: mart.callbacks.AttackInEvalMode
+  module_classes: ???
```
Comment (dxoigmn) [inline mart/configs/callbacks/attack_in_eval_mode.yaml:3]: Add support for `module_names`?
